### PR TITLE
[Feature] Add Qwen3.5 Model Support for DFlash

### PR DESCRIPTION
**Pull request by EanWang211123**
## Motivation
Add DFlash speculative decoding support for Qwen3.5 models, following the implementation approach in https://github.com/sgl-project/sglang/pull/16818

## Modifications
Similar to https://github.com/sgl-project/sglang/pull/18387:
- In `dflash_worker.py`, save the global server_args before creating the draft worker and restore it afterward
- Added `set_dflash_layers_to_capture` interface for better layer management

## Tests
Successfully trained and tested with Qwen3.5-27B model using DFlash in [SpecForge](https://github.com/sgl-project/SpecForge):
- Dataset: eaglechat
- Training steps: 6 epochs with 8k data points

**Test Environment:**
- Hardware: H100 (4x GPUs)
- Tensor Parallel size: 4
- Repository: https://github.com/z-lab/dflash

**Test Command:**
```bash
SGLANG_DISABLE_CUDNN_CHECK=1 \
CUDA_VISIBLE_DEVICES=4,5,6,7 \
python benchmark_sglang.py \
--tp-size 4 \
--target-model /models/Qwen/Qwen3___5-27B/ \
--draft-model /draft-model-path/ \
--concurrencies  1,8,32 \
--dataset-name humaneval  \
--attention-backends fa3 \
--max-running-requests 32
```

#### Baseline Output (tok/s)

| concurrencies | 1 | 4 | 32 |
|---|---|---|---|
| tok/s | 103.61 | 361.46 | 1,860.84 |

#### DFLASH Output (tok/s)

| concurrencies | 1 | 4 | 32 |
|---|---|---|---|
| tok/s | 207.19 | 684.67 | 2,115.39 |

#### Speedup (DFLASH / Baseline)

| concurrencies | 1 | 4 | 32 |
|---|---|---|---|
| speedup | 2.000x | 1.894x | 1.137x |

#### DFLASH Acceptance Length

| concurrencies | 1 | 4 | 32 |
|---|---|---|---|
| acceptance length | 3.233 | 3.216 | 3.219 |

## Summary
The implementation shows significant performance improvements with DFlash enabled, achieving up to 2x speedup for single concurrent requests and maintaining good acceleration across different concurrency levels.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Comment by liyucheng09** [comment]
@EanWang211123 how's the `accept len:` in sglang log? I tried to reproduce your results but mine number is rather low.
image

**Comment by moehanabi** [comment]
Update on 4.7:
the reason seems that this version of sglang will output repeatedly which causes the draft not accepted. It works well after merge newest main commits

Update on 4.8:
have create a pr at https://github.com/EanWang211123/sglang/pull/2
merged from v0.5.10 tag & fix memory_pool_config


Reply at first time:
Hi! Thanks for your great work!
For qwen3 30b a3b:
I met the same low accept len result, with my own 150k train data.
and I found the first accept len was high (still lower than eagle3), then it will drop after following requests, but the test data items are all similar (and work well in eagle3)
accept len

use TP=2 to ensure enough kvcache space:

accept len
